segmentation/utils/spatial_analysis.py

```diff
@@ -88,16 +91,17 @@ def calculate_channel_spatial_enrichment(dist_matrix, marker_thresholds, all_pat
 
     for point in points:
         # Subsetting expression matrix to only include patients with correct label
-        patient_ids = all_patient_data.iloc[:, patient_idx] == point
-        patient_data = all_patient_data[patient_ids]
+        # patient_ids = all_patient_data.iloc[:, patient_idx] == point
+        # patient_data = all_patient_data[patient_ids]
         # Patients with correct label, and only columns of markers
-        patient_data_markers = data_markers[patient_ids]
+        # patient_data_markers = data_markers[patient_ids]
         # Subsetting the column with the cell labels
-        label_idx = patient_data.iloc[:, cell_label_idx]
+        # label_idx = patient_data.iloc[:, cell_label_idx]
 
         # Get close_num and close_num_rand
         close_num, marker1_num, marker2_num = spatial_analysis_utils.compute_close_cell_num(
-            patient_data_markers, label_idx, thresh_vec, dist_matrix, marker_num, dist_lim)
+            dist_mat=dist_matrix, dist_lim=100, num=marker_num, analysis_type="Threshold", point=point,
+            all_patient_data=all_patient_data, data_markers=data_markers, thresh_vec=thresh_vec)
```

Comment:
This looks much cleaner already! Right now, we're passing all_patient_data, which is the entire matrix. We're also passing in data_markers, which is the entire matrix, but only with the marker columns. This is a bit redundant, is there a way we can further simplify this so that we're only passing in the data once?